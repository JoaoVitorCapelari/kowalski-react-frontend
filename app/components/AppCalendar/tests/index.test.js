/**
 * Testing our Button component
 */

import React from 'react';
import { shallow } from 'enzyme';
import startOfWeek from 'date-fns/start_of_week';
import endOfWeek from 'date-fns/end_of_week';

import { AppCalendar } from '../index';
import { dummyIntlObject } from '../../../support/tests/stub';

const renderComponent = (props = {}) => (
  shallow(<AppCalendar {...{ ...props, ...dummyIntlObject }} />)
);

describe('<AppCalendar />', () => {
  it('renders', () => {
    const date = new Date(2014, 1, 1);
    const tooltips = {};
    tooltips[new Date(2014, 1, 2)] = 'Testing tooltip';
    const renderedComponent = renderComponent({
      selectedDate: new Date(2014, 1, 1),
      options: {
        highlightedRanges: [
          [
            startOfWeek(date), endOfWeek(date),
          ],
        ],
        tooltips,
        disabledDates: [new Date(2014, 1, 5)],
      },
    });
    expect(renderedComponent.html()).toMatchSnapshot();
  });

  it('callbacks onDateClicked properly', () => {
    const onDateClicked = jest.fn();

    const renderedComponent = renderComponent({
      selectedDate: new Date(2014, 1, 1),
      onDateClicked,
    });

    const dayElement = renderedComponent.find('button').findWhere((node) => node.text() === '20').first();
    dayElement.simulate('click');
    expect(onDateClicked.mock.calls.length).toBe(1);
    expect(onDateClicked.mock.calls[0][0]).toEqual(new Date(2014, 1, 20));
  });

  it('callbacks onPreviousMonthClicked properly', () => {
    const onPreviousMonthClicked = jest.fn();

    const renderedComponent = renderComponent({
      selectedDate: new Date(2014, 1, 1),
      onPreviousMonthClicked,
    });

    const previousMonthBtn = renderedComponent.find('div.calendar-nav-left').find('button').first();
    previousMonthBtn.simulate('click');
    expect(onPreviousMonthClicked.mock.calls.length).toBe(1);
    expect(onPreviousMonthClicked.mock.calls[0][0]).toEqual(undefined);
  });

  it('callbacks onNextMonthClicked properly', () => {
    const onNextMonthClicked = jest.fn();

    const renderedComponent = renderComponent({
      selectedDate: new Date(2014, 1, 1),
      onNextMonthClicked,
    });

    const previousMonthBtn = renderedComponent.find('div.calendar-nav-right').find('button').first();
    previousMonthBtn.simulate('click');
    expect(onNextMonthClicked.mock.calls.length).toBe(1);
    expect(onNextMonthClicked.mock.calls[0][0]).toEqual(undefined);
  });
});
