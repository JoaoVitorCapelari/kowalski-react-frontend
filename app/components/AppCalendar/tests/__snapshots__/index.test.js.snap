// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppCalendar /> renders 1`] = `"<div class=\\"calendar\\"><div class=\\"calendar-nav\\"><div class=\\"calendar-nav-left\\"><button class=\\"button is-text\\"><svg fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 40 40\\" style=\\"vertical-align:middle;\\"><g><path d=\\"m31.1 6.7l-11.8 11.9 11.8 11.8q0.5 0.4 0.5 1t-0.5 1l-3.7 3.7q-0.4 0.5-1 0.5t-1-0.5l-16.5-16.5q-0.5-0.4-0.5-1t0.5-1l16.5-16.6q0.5-0.4 1-0.4t1 0.4l3.7 3.7q0.5 0.4 0.5 1t-0.5 1z\\"></path></g></svg></button></div><div class=\\"yearMonth\\">Object Object</div><div class=\\"calendar-nav-right\\"><button class=\\"button is-text\\"><svg fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 40 40\\" style=\\"vertical-align:middle;\\"><g><path d=\\"m31.2 19.6l-16.6 16.5q-0.4 0.5-1 0.5t-1-0.5l-3.7-3.7q-0.4-0.4-0.4-1t0.4-1l11.9-11.8-11.9-11.9q-0.4-0.4-0.4-1t0.4-1l3.7-3.7q0.5-0.4 1-0.4t1 0.4l16.6 16.6q0.4 0.4 0.4 1t-0.4 1z\\"></path></g></svg></button></div></div><div class=\\"calendar-container\\"><div class=\\"calendar-header\\"><div class=\\"calendar-date\\">Sun</div><div class=\\"calendar-date\\">Mon</div><div class=\\"calendar-date\\">Tue</div><div class=\\"calendar-date\\">Wed</div><div class=\\"calendar-date\\">Thu</div><div class=\\"calendar-date\\">Fri</div><div class=\\"calendar-date\\">Sat</div></div><div class=\\"calendar-body\\"><div class=\\"calendar-date  is-disabled calendar-range range-start\\"><button class=\\"date-item  is-active\\">26</button></div><div class=\\"calendar-date  is-disabled calendar-range\\"><button class=\\"date-item \\">27</button></div><div class=\\"calendar-date  is-disabled calendar-range\\"><button class=\\"date-item \\">28</button></div><div class=\\"calendar-date  is-disabled calendar-range\\"><button class=\\"date-item \\">29</button></div><div class=\\"calendar-date  is-disabled calendar-range\\"><button class=\\"date-item \\">30</button></div><div class=\\"calendar-date  is-disabled calendar-range\\"><button class=\\"date-item \\">31</button></div><div class=\\"calendar-date  calendar-range range-end\\"><button class=\\"date-item  is-active is-active\\">1</button></div><div class=\\"calendar-date  tooltip\\" data-tooltip=\\"Testing tooltip\\"><button class=\\"date-item \\">2</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">3</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">4</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">5</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">6</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">7</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">8</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">9</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">10</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">11</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">12</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">13</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">14</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">15</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">16</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">17</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">18</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">19</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">20</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">21</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">22</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">23</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">24</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">25</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">26</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">27</button></div><div class=\\"calendar-date \\"><button class=\\"date-item \\">28</button></div><div class=\\"calendar-date  is-disabled\\"><button class=\\"date-item \\">1</button></div></div></div></div>"`;
